users.py

```diff
@@ -0,0 +1,113 @@
+"""User Signup and Signin class Logic"""
+
+from constant import (
+    ROLE,
+    ROLE_HOST,
+    ROLE_WATCHER,
+    USER_JSON,
+    USER_EXIST,
+    USER_NOT_EXIST,
+    WELCOME,
+    AUTHENTICATE_ERROR,
+    DATA_JSON_PATH,
+)
+from exceptions import AuthenticationError, UnderAge
+from utils import (
+    custom_print_msg,
+    read_json,
+    update_main_theater_data,
+    user_input,
+    user_input_int_type,
+    password_hash,
+)
+
+
+class User:
+    """__main__ class to implement user signup & signin process"""
+
+    def __init__(self, username: str) -> None:
+        """
+        user class Constructor
+        @param username:str
+        @param password:int
+        """
+        self.username = username
+        self.is_user_exist()
+
+    def sign_up(self):
+        """
+        method to implement signup process
+        @return:
+        """
+        custom_print_msg("Please Enter User Information")
+        password = password_hash(input("Enter Your Password: "))
+        role = user_input_int_type(f"Please Select User Role {ROLE}: ")
+        if int(role) == 1:
```

Comment:
`
if role not in [ROLE_HOST, ROLE_WATCHER]:
     raise
role = int(user_input_int_type(f"Please Select User Role {ROLE}: "))
no need if else conditions